src/bump/updateRelatedChangeType.ts

```diff
@@ -9,50 +9,80 @@ import { BumpInfo } from '../types/BumpInfo';
  */
 export function updateRelatedChangeType(
   pkgName: string,
-  changeType: ChangeType,
+  changeType: ChangeInfo,
   bumpInfo: BumpInfo,
+  dependentChangeInfos: Map<string, Array<ChangeInfo>>,
   bumpDeps: boolean
 ) {
   const { packageChangeTypes, packageGroups, dependents, packageInfos, dependentChangeTypes, groupOptions } = bumpInfo;
 
-  const disallowedChangeTypes = packageInfos[pkgName].combinedOptions?.disallowedChangeTypes ?? [];
+  const packageInfo = packageInfos[pkgName];
+  const disallowedChangeTypes = packageInfo.combinedOptions?.disallowedChangeTypes ?? [];
+
+  let depChangeType = {
```

Comment:
maybe cleaner to have getMaxChangeType change its purpose to "updateChangeInfoWithMaxType"